### PR TITLE
Remove default libcurl Content-Type header for empty POST requests

### LINKED ISSUE #20734: Httpclient PostAsync with null HttpContent differs on Windows/Linux
```
var client = new HttpClient();
var result = client.PostAsync("http://localhost", null).Result.Content.ReadAsStringAsync().Result;
Console.WriteLine("Content-type:" + result);
```
Only thing server-side does is echo back the "content-type" header.

**When runned on windows 10, this is what I get:**
`Content-type:`

**When runned on Ubuntu, I instead get this:**
`Content-type:application/x-www-form-urlencoded
`

Is this the intended behaviour?

### PR DESCRIPTION
**Pull request by khellang**
Fixes #20734

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs [hunk 1]
```diff
@@ -173,7 +173,23 @@ public PostScenarioTest(ITestOutputHelper output)
             var credential = new NetworkCredential(UserName, Password);
             await PostUsingAuthHelper(serverUri, ExpectedContent, content, credential, preAuthenticate: true);
         }
-        
+
+        [OuterLoop] // TODO: Issue #11345
+        [Theory, MemberData(nameof(EchoServers))]
+        public async Task PostNoContent_ExcludeContentTypeHeader_Success(Uri serverUri)
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs]: The test should be renamed.  "ExcludeContentTypeHeader" is not the condition being tested but rather the expected result.

I would suggest the test be named something like "PostAsync_EmptyContent_ContentTypeHeaderNotSent"

Comment (khellang) [inline src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs]: Yeah, I didn't quite grok the naming scheme here 😛

**Comment by khellang** [comment]
A couple of things regarding the test...
1. I added the `OuterLoopAttribute`, like the other tests. I see that #11345 (and #14519) has been closed. Does that mean that these attributes should be removed now?
2. Since I don't have a UNIX machine to test on, I'd hoped I could get the test to fail on the CI machine, but since these tests are outer loop, I'm guessing it won't run it, or?

**Comment by stephentoub** [comment]
@dotnet-bot help

**Comment by stephentoub** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu14.04 Debug
@dotnet-bot test outerloop netcoreapp OSX10.12 Debug

**Comment by davidsh** [comment]
>Does that mean that these attributes could/should be removed now?

No. Please leave the OuterLoop attributes on the tests.

File: src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs]: >the naming scheme here 

This the book we recommend. The naming scheme comes from here:
https://smile.amazon.com/Art-Unit-Testing-examples/dp/1617290890/ref=sr_1_1?ie=UTF8&qid=1496771523&sr=8-1&keywords=art+of+unit+testing

**Comment by khellang** [comment]
@dotnet-bot test innerloop Ubuntu14.04 Debug

**Comment by khellang** [comment]
@dotnet-bot test Linux x64 Tests - Debug - Ubuntu.1404.Amd64.Open

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs [hunk 2]
```diff
@@ -713,6 +713,11 @@ internal void SetRequestHeaders()
                         ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoContentType));
                     }
                 }
+                else
+                {
+                    // Remove the Content-Type header libcurl adds by default.
+                    ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoContentType));
+                }
```
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs]: This is ok, but it'd be nice if possible to consolidate the logic of actually adding the header, e.g. something like:
```C#
bool suppressContentType = true;
if (_requestMessage.Content != null)
{
    AddRequestHeaders(_requestMessage.Content.Headers, slist);
    suppressContentType = (_requestMessage.Content.Headers.ContentType == null);
}

if (suppressContentType)
{
    // Remove the Content-Type header libcurl adds by default)
    ThrowOOMIfFalse(Interop.Http.SListAppend(slist, NoContentType);
}
```

**Comment by khellang** [comment]
@dotnet-bot test Portable Linux x64 Debug Build please

File: src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs [hunk 3]
```diff
@@ -173,7 +173,23 @@ public PostScenarioTest(ITestOutputHelper output)
...
             await PostUsingAuthHelper(serverUri, ExpectedContent, content, credential, preAuthenticate: true);
         }
-        
+
+        [OuterLoop] // TODO: Issue #11345
+        [Theory, MemberData(nameof(EchoServers))]
+        public async Task PostAsync_EmptyContent_ContentTypeHeaderNotSent(Uri serverUri)
+        {
+            using (var client = new HttpClient())
+            {
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs]: Nit: extra level of braces/indentation isn't necessary

**Comment by khellang** [comment]
Not really sure how to rerun **Linux x64 Tests - Debug - Ubuntu.1404.Amd64.Open** 😕

File: src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs (hunk 3 above)
Comment (khellang) [inline src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs]: :+1: I wasn't really sure about the stuping here.

**Comment by stephentoub** [comment]
> Not really sure 

@mmitche can correct me if I'm wrong, but I believe the Portable Linux x64 Debug Build leg will re-kick off all of the Linux x64 Tests legs when it completes the build.  My understanding is that right now it's not possible to just re-run an individual test leg.

**Comment by mmitche** [comment]
@dotnet/eng-first for visibility @stephentoub Yes correct. Currently, this is not individually addressable.

**Comment by khellang** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu14.04 Debug
@dotnet-bot test outerloop netcoreapp OSX10.12 Debug

**Comment by stephentoub** [comment]
@mmitche, we're still seeing this deletion failure errors (build/tests complete successfully)... is this still expected?
https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netcoreapp_osx_debug_prtest/49/
```
Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:41:02.83
Command execution succeeded.
[outerloop_netcoreapp_osx_debug_prtest] $ /bin/sh -xe /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/hudson4363441577097059432.sh
+ tar -czf bin/build.tar.gz --directory=bin/runtime/netcoreapp-OSX-Debug-x64 .
[xUnit] [INFO] - Starting to record.
[xUnit] [INFO] - Processing xUnit.Net-v2 (default)
[xUnit] [INFO] - [xUnit.Net-v2 (default)] - 206 test report file(s) were found with the pattern 'bin/**/testResults.xml' relative to '/Users/dotnet-bot/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_osx_debug_prtest' for the testing framework 'xUnit.Net-v2 (default)'.
[xUnit] [ERROR] - The plugin hasn't been performed correctly: Problem on deletion
```

**Comment by stephentoub** [comment]
The Linux failure is unrelated:
https://github.com/dotnet/corefx/issues/17925

**Comment by khellang** [comment]
:heart::heart:

**Comment by stephentoub** [comment]
Thanks, @khellang.

**Comment by khellang** [comment]
I must say the contributing story for corefx (clone, setup, build, etc.) has come a *long* way since the last time I contributed. Really smooth. Kudos to you! :sparkles:

**Comment by karelz** [comment]
Thanks @khellang for your feedback! Can you post some specific links you really liked? Docs on wiki? Or the "old" one in Documentation directory? (it will help us drive in the right direction)

cc @haralabidis who helped a lot with most of the new contribution docs on wiki.

**Comment by khellang** [comment]
I just went to CONTRIBUTING.md, skimmed through it while cloning the repo, ran build.cmd, went for a hair cut, came back, opened the System.Net.Http sln in VS and started hacking :wink: No fuss - it just worked :+1: Also tried running the `RebuildAndTest` target a few times from the CLI. Worked beautifully. The CI system is also pretty impressive, albeit a bit confusing (see above).

**Pull request merged**